src/misceval.js

```diff
@@ -894,7 +894,11 @@ Sk.misceval.asyncToPromise = function(suspendablefn, suspHandlers) {
                 try {
                     // jsh*nt insists these be defined outside the loop
                     var resume = function() {
-                        handleResponse(r.resume());
+                        try {
+                            handleResponse(r.resume());
+                        } catch (e) {
+                            reject(e);
```

Comment:
@albertjan I'm seeing the error in the screen shot below in a specific scenario. I don't know how reproducible it'll be but here are the steps.

* Run the program below. (It's a slight modification of an example from @waywaaard)
* Move the pixel around
* Stop the program
* Run the program again
* Move the pixel around once << this is when the error is thrown - I also noticed that the pixel briefly moved back to the last location before appearing at it's starting location

![screenshot 2016-07-28 at 8 55 31 pm](https://cloud.githubusercontent.com/assets/688178/17234322/bb7cc132-5505-11e6-96b7-49d263e4b6e3.png)

```python
from sense_hat import SenseHat, ACTION_PRESSED, ACTION_RELEASED
#from signal import pause

x = 3
y = 3
sense = SenseHat()

def clamp(value, min_value=0, max_value=7):
    return min(max_value, max(min_value, value))

def pushed_up(event):
    global y
    if event.action != ACTION_RELEASED:
        y = clamp(y - 1)

def pushed_down(event):
    global y
    if event.action != ACTION_RELEASED:
        y = clamp(y + 1)

def pushed_left(event):
    global x
    if event.action != ACTION_RELEASED:
        x = clamp(x - 1)

def pushed_right(event):
    global x
    if event.action != ACTION_RELEASED:
        x = clamp(x + 1)

def refresh():
    sense.clear()
    sense.set_pixel(x, y, 255, 0, 0)

sense.stick.direction_up = pushed_up
sense.stick.direction_down = pushed_down
sense.stick.direction_left = pushed_left
sense.stick.direction_right = pushed_right
sense.stick.direction_any = refresh
refresh()

while True:
    pass

# This seems not to be possible in skulpt: https://docs.python.org/2/library/signal.html#signal.pause
#pause()
```